routers/web/repo/view.go

```diff
@@ -1045,7 +1045,7 @@ func renderHomeCode(ctx *context.Context) {
 
 		if !opts.Repo.IsMirror && !opts.BaseRepo.IsMirror &&
 			opts.BaseRepo.UnitEnabled(ctx, unit_model.TypePullRequests) &&
-			baseRepoPerm.CanRead(unit_model.TypePullRequests) {
+			baseRepoPerm.CanWrite(unit_model.TypePullRequests) {
```

Comment:
IIRC "read" means "create", "write" means "delete/modify"

Reply:
I tested it:
- user A create a public repo
- user B access this repo, the pull request button is disabled.
- add user B as a contributor with write permission
- the pull request button is enabled.
- convert the permission to read
- the pull request button is disabled.

It seems that this has been changed.

Reply:
sorry, that is because user B has no branch to create PR